pkg/kubelet/container/helpers_test.go

```diff
@@ -481,17 +481,19 @@ func TestMakePortMappings(t *testing.T) {
 				Name: "fooContainer",
 				Ports: []v1.ContainerPort{
 					port("", v1.ProtocolTCP, 80, 8080, "127.0.0.1"),
-					port("", v1.ProtocolTCP, 443, 4343, "192.168.0.1"),
+					port("", v1.ProtocolUDP, 80, 8080, "127.0.0.1"),
+					port("", v1.ProtocolTCP, 80, 0, "127.0.0.1"),
+					port("", v1.ProtocolTCP, 80, 8888, ""),
 					port("foo", v1.ProtocolUDP, 555, 5555, ""),
 					// Duplicated, should be ignored.
 					port("foo", v1.ProtocolUDP, 888, 8888, ""),
-					// Duplicated, should be ignored.
```

Comment:
I'm a little hesitant deleting this test case, as I presume it was added because it was intended that duplicate container ports should be ignored.

At a higher level, this makes me wonder if the error message you got was not the result of a bug, but rather intended functionality. Can you share more about why you expected to be able to bind the same container port to two different host ports?

Reply:
Multiple hostports already work as expected when manually specifying different port names. It's just the automatic port name generation that's broken.

> Can you share more about why you expected to be able to bind the same container port to two different host ports?

For example when doing a migration from one port to the other.

Reply:
I agree that removing this test case is slightly concerning. would like an ack on this from @kubernetes/sig-network-pr-reviews